datalad/support/annexrepo.py

```diff
@@ -3476,24 +3476,43 @@ def _save_post(self, message, status, partial_commit):
         # first do standard GitRepo business
         super(AnnexRepo, self)._save_post(
             message, status, partial_commit)
-        # check if we have to deal with adjusted branch
+        self.sync_corresponding_branch()
+
+    def sync_corresponding_branch(self, remote=None):
+        """Sync a managed branch with correspoding branch with `git-annex sync`
+
+        If the active branch is not a managed branch, this method will silently
+        skip any synchronization.
+
+        Parameters
+        ----------
+        remote : str, optional
+          If given, it is passed to the `git annex sync` call.
+        """
         branch = self.get_active_branch()
-        adjusted_match = ADJUSTED_BRANCH_EXPR.match(branch if branch else '')
-        if adjusted_match:
-            orig_branch = adjusted_match.group('name')
-            synced_branch = 'synced/{}'.format(orig_branch)
-            had_synced_branch = synced_branch in self.get_branches()
-            lgr.debug(
-                "Git-annex adjusted branch detected, sync'ing %s",
-                orig_branch)
+        corresponding_branch = self.get_corresponding_branch(branch)
+        if branch != corresponding_branch:
```

Comment:
Shouldn't this be `==`?

Reply:
Yes, and another conditional was also reversed :man_facepalming: now fixed.